### PR TITLE
Cluedo 

### PR DESCRIPTION
**Pull request by carolinebirgersson**
Here is my Cluedo page!

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -1,35 +1,158 @@
...
+    name: 'Poison',
+    color: 'lightgrey'
+  }
 
+  const axe = { 
+    name: 'Axe',
+    color: 'lightpurple',
+    image: 'assets/axe.jpg'
+    //I would have added pictures for all of the weapons if I had the time.
+  }
```
Comment (ssundblom) [inline code/clue.js]: Really like the comments you left here :)

File: code/clue.js [hunk 2]
```diff
@@ -1,35 +1,158 @@
...
+    color: 'saffron'
+  }
+
+  const pistol = {
+    name: 'Pistol',
+    color: 'black'
+  }
+
+// OBJECTS FOR ROOMS  
+  //I would have added objects for all of the rooms if I had the time.
```
Comment (ssundblom) [inline code/clue.js]: Here as well!

File: code/clue.js [hunk 3]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
```
Comment (ssundblom) [inline code/clue.js:163]: Maybe remove the comments from the assignment? Makes it a bit easier to read :)

File: code/clue.js [hunk 4]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
+const revealMystery = (mystery) => {
+  event.preventDefault();
+  document.getElementById('solutionText').innerText = `The murder was comitted by ${missScarlet.firstName} ${missScarlet.lastName} with an ${axe.name}, in the ${rooms[4]}.`
+  //Makes the solution-pictures go to the cards when clicking. 
+  //How to fix the "bug": when you click on a card first - and then click on the button - the name of the previous killer card stands there instead of the name that belongs to the chosen killer.  
+  killerImage.src = missScarlet.image
+  roomImage.src = library.image
+  weaponImage.src = axe.image
+}
+revealMystery()
```
Comment (ssundblom) [inline code/clue.js]: I think this is why the scarlet picture, the library image and axe always is revealed when clicking 'crime solution'. It's hard coded to be these specific pictures. Could it be better to give the 'solutionText' something like ${mystery.killer.firstName}, ${mystery.killer.lastName},  ${mystery.weapon} and so on?

File: code/clue.js [hunk 5]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
+  mystery.room = randomSelector(rooms)
+  document.getElementById(
+    'roomName'
+  ).innerHTML = `${mystery.room}`
+ }
+
+ document.getElementById(
+  'roomCard'
+  ).addEventListener('click', pickRoom);
+
```
Comment (ssundblom) [inline code/clue.js]: love your functions! Very nice and easy to read!

File: code/index.html [hunk 6]
```diff
@@ -1,37 +1,43 @@
...
-      <article class="room-deck" id="roomCard">
+      <article class='room-deck' id='roomCard'>
         <p>ROOM</p>
-        <p id="roomName"></p>
+        <p id='roomName'></p>
+        <img id='roomImage' src='' height='75%'>
       </article>
+      <article class="solution-container">
+      <button onclick='revealMystery()' id="mystery">CRIME SOLUTION</button>
+      <div class='solution'>
```
Comment (ssundblom) [inline code/index.html:35]: Here I see you've used onclick and in the JS code I saw you used addEventListener! Super that you master and understand both of them :) 😍 Maybe nest time choose one of them for a more unified code ☺️

File: code/style.css [hunk 7]
```diff
@@ -1,58 +1,250 @@
...
-  font-family: "Courier New", Courier, monospace;
+  font-family:'Permanent Marker', cursive;
+  background-color: #3D915D;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+  background-image: url(https://i.pinimg.com/564x/e2/ae/f6/e2aef657b2faac87de8c27ee392bc5ca.jpg);
+}
+
```
Comment (ssundblom) [inline code/style.css:19]: The styling 😍 
I love the color theme and the newspaper background - very murder mysterious!

File: code/style.css [hunk 8]
```diff
@@ -1,58 +1,250 @@
...
+h1 {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  font-family: 'Permanent Marker';
+  background: white;
```
Comment (ssundblom) [inline code/style.css]: I don't think you need to repeat the font-family in the classes when you have them in the body.  Makes a little less work 👍

File: code/style.css [hunk 9]
```diff
@@ -1,58 +1,250 @@
...
+  box-shadow: 5px 10px 8px #888888;
   padding-top: 30px;
   border-radius: 16px;
+  width: 250px;
+  height: 400px;
+}
+
+.killer-deck:hover, .weapon-deck:hover, .room-deck:hover {
+  background:#9A2931;
 }
```
Comment (ssundblom) [inline code/style.css]: The hover effect is so nice! Love that detail 👌

File: code/style.css [hunk 10]
```diff
@@ -1,58 +1,250 @@
...
   margin: 0 auto;
+  font-size: 20px;
+  border-radius: 8px;
+  padding: 25px;
+}
+
+button:hover {
+  background:#9A2931;
+}
+
```
Comment (ssundblom) [inline code/style.css:118]: The hover here as well! :)

**Review by ssundblom — CHANGES_REQUESTED** [review]
Good job Caroline! I think you've done a great game and I really love the styling and all the details you put in to it 😍
Now I just think you need to get the 'crime solution' button to work and you are good to go! Celebrate your progress! 🥳

***change request***
- You should be able to reveal the mystery.

File: code/clue.js [hunk 11]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  suspects,
+  weapons,
+  rooms:[]
```
Comment (EmOlsz) [inline code/clue.js]: Why do you initialise your mystery object with arrays of all suspects, weapons and empty array of rooms?
At the beginning it should be empty, and later on you should add to each property (so, suspect, weapon and rooms (please, notice that these are single, not multi)) value as single element from suspets, weapons and rooms array. So

1. Beginning :
```
const mystery = {
  suspect: null,
  weapon: null,
  rooms: null
}
```

2. Finish :
```
mystery = {
  suspect: //Mr green object,
  weapon: // knife object,
  rooms: // dining room object
}
```

File: code/clue.js [hunk 12]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
+}
 
+//Killer
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
+
   mystery.killer = randomSelector(suspects)
```
Comment (EmOlsz) [inline code/clue.js]: So, `suspect` or a `killer`? You haven't initialised your object with property called `killer` so after this line your mystery object has 4 properties : suspects, weapons, rooms and killer. That doesn't make sense

File: code/clue.js [hunk 13]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
+    ).style.background = mystery.killer.color
+  }
+
+  document.getElementById(
+    'killerCard'
+    ).addEventListener('click', pickKiller);
+
+//Weapons
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
```
Comment (EmOlsz) [inline code/clue.js]: Another new property added to mystery object, right now we have 5 of them : suspects, weapons, rooms, killer and weapon

File: code/clue.js [hunk 14]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById(
+  'weaponCard'
+  ).addEventListener('click', pickWeapon);
+
+//Rooms
+ const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
```
Comment (EmOlsz) [inline code/clue.js]: 6 of them right now

File: code/clue.js [hunk 15]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
+  ).innerHTML = `${mystery.room}`
+ }
+
+ document.getElementById(
+  'roomCard'
+  ).addEventListener('click', pickRoom);
+
+// Reveal the mystery
+const revealMystery = (mystery) => {
+  event.preventDefault();
```
Comment (EmOlsz) [inline code/clue.js]: `event` is no variable or some sort of that. It's an argument passed to your function that you need to provide. In your code, you called it `mystery`, so you should write `mystery.preventDefault()`

File: code/clue.js (hunk 4 above)
Comment (EmOlsz) [reply code/clue.js]: Yep, it should be done like this

File: code/index.html [hunk 16]
```diff
@@ -1,37 +1,43 @@
...
+<html lang='en'>
   <head>
-    <meta charset="UTF-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
-    <link rel="stylesheet" href="style.css" />
+    <meta charset='UTF-8' />
+    <meta name='viewport' content='width=device-width, initial-scale=1.0' />
+    <meta http-equiv='X-UA-Compatible' content='ie=edge' />
+    <link rel='stylesheet' href='style.css' />
```
Comment (EmOlsz) [inline code/index.html:7]: Let's not forget about `./` in path to file

File: code/index.html [hunk 17]
```diff
@@ -1,37 +1,43 @@
...
         <p>KILLER</p>
-        <p id="killerName"></p>
-        <p id="killerAge"></p>
-        <p id="killerOccupation"></p>
-        <img id="killerImage" src="" height="100" />
-        <p id="killerDescription"></p>
+        <p id='killerName'></p>
+        <p id='killerAge'></p>
+        <p id='killerOccupation'></p>
+        <img id='killerImage' src='' height='75%' />
```
Comment (EmOlsz) [inline code/index.html:19]: We should avoid setting style properties via HTML - all of them should be added in CSS files

File: code/index.html [hunk 18]
```diff
@@ -1,37 +1,43 @@
...
+      <footer><p class="footer-text">© Cluedo by Caroline Birgersson</p></footer>
+    </article>
     </section>
-
-    <h1 id="mystery"></h1>
-
-    <button>REVEAL THE CRIME</button>
-
-    <script src="clue.js"></script>
+    <script src='clue.js'></script>
```
Comment (EmOlsz) [inline code/index.html:41]: `./`

File: code/style.css (hunk 8 above)
Comment (EmOlsz) [reply code/style.css]: Good point Sara!

File: code/style.css [hunk 19]
```diff
@@ -1,58 +1,250 @@
...
   align-items: center;
+  cursor: pointer;
+  margin: 50px auto;
+  border-radius: 25px;
+  width: 80%;
+  height: 70%;
+  background: rgb(10, 53, 10);
+}
+
+#weaponName, #killerName, #roomName {
```
Comment (EmOlsz) [inline code/style.css]: We should avoid styling by id's, use classes instead

File: code/style.css [hunk 20]
```diff
@@ -1,58 +1,250 @@
...
   border-radius: 16px;
+  width: 250px;
+  height: 400px;
+}
+
+.killer-deck:hover, .weapon-deck:hover, .room-deck:hover {
+  background:#9A2931;
 }
 
 .weapon-deck {
```
Comment (EmOlsz) [inline code/style.css:69]: There is a lot of shared styles between killer-deck, weapon-deck and room-deck - They could be unified to one class, and have separate classes only for styles that are not shared between them

File: code/style.css [hunk 21]
```diff
@@ -1,58 +1,250 @@
...
+  flex-direction: column;
+  align-items: center;
+  padding-top: 100px;
+  padding-bottom: 80px;
+  width: 100%;
+}
+
+.solution {
+  text-align: center;
+  content: '';
```
Comment (EmOlsz) [inline code/style.css]: What do you use `content` property for? This property is used only for `after` and `before` elements, it doesn't have any impact on styles over here

File: code/clue.js [hunk 22]
```diff
@@ -40,20 +163,68 @@ const randomSelector = array => {
...
     'killerName'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById(
+    'killerCard'
+    ).style.background = mystery.killer.color
+  }
+
+  document.getElementById(
+    'killerCard'
+    ).addEventListener('click', pickKiller);
```
Comment (EmOlsz) [inline code/clue.js]: Very nice :)

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Hi Caroline! I would like to congratulate you on effort you put into this project, but unfortunately there are a lot of bugs in your code...
I got impression that you haven't checked your code before submitting it. I would love if you could analyse it step by step (at least JS file) and deduct what's wrong with it. Hope my comments will help you with that :)
I will need to see changes before I will be able to approve it.

Let me point out what should be fixed :
1. You initialise your project in wrong way, adding new properties later on. Please, check out my comments in code regarding that
2. You use `event` object and default function parameters wrong way
3. Your solution is hardcoded, so we always reveal mystery with the same person, weapon and room
4. I've got impression that you don't understand some of style you use. Please, if you don't know what they are doing, learn about them or delete them

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**